src/Elasticsearch.Extensions.Logging/ElasticsearchLogger.cs

```diff
@@ -61,50 +62,103 @@ public class ElasticsearchLogger : ILogger
 
 		private void AddScopeValues(LogEvent logEvent)
 		{
-			if (_options.IncludeScopes)
+			if (!_options.IncludeScopes) return;
+
+			void AddScopeValue<TState>(TState scope, LogEvent log)
 			{
-				_scopeProvider?.ForEachScope((scope, le) =>
-				{
-					le.Labels ??= new Labels();
-					le.Scopes ??= new List<string>();
-
-					var isFormattedLogValues = false;
-					if (scope is IEnumerable<KeyValuePair<string, object>> scopeValues)
-					{
-						foreach (var kvp in scopeValues)
-						{
-							if (kvp.Key == "{OriginalFormat}")
-							{
-								isFormattedLogValues = true;
-								continue;
-							}
-							le.AssignField(kvp.Key, FormatValue(kvp.Value));
-						}
-					}
-
-					var formattedScope = isFormattedLogValues ? scope.ToString() : FormatValue(scope);
-					le.Scopes.Add(formattedScope);
-				}, logEvent);
+				if (scope is null) return;
+				log.Labels ??= new Labels();
+				log.Scopes ??= new List<string>();
+
+
+				var scopeValues = (scope as IEnumerable<KeyValuePair<string, object>>)?.ToList();
+				var scopeName = scope as string ?? scope.GetType().Name;
+				if (scopeValues != null && scopeValues.Any(kv=>kv.Key == "{OriginalFormat}"))
+					scopeName = FormatValue(scope);
+				log.Scopes.Add(scopeName);
+
+				if (scopeValues == null) return;
+
+				foreach (var kvp in scopeValues)
+					AssignStateOrScopeLabels(logEvent, kvp);
 			}
+
+			_scopeProvider?.ForEachScope((o, @event) => AddScopeValue(o, @event), logEvent);
 		}
 
 		private void AddStateValues<TState>(TState state, LogEvent logEvent)
 		{
-			if (state is IEnumerable<KeyValuePair<string, object>> stateValues)
+			var stateValues = state as IEnumerable<KeyValuePair<string, object>>;
+			if (stateValues == null) return;
+
+			foreach (var kvp in stateValues)
+				AssignStateOrScopeLabels(logEvent, kvp);
+		}
+
+		private void AssignStateOrScopeLabels(LogEvent logEvent, KeyValuePair<string, object> kvp)
+		{
+			if (kvp.Key == "{OriginalFormat}")
 			{
-				foreach (var kvp in stateValues)
-				{
-					if (kvp.Key == "{OriginalFormat}")
-					{
-						logEvent.MessageTemplate = kvp.Value.ToString();
-						continue;
-					}
-					logEvent.AssignField(kvp.Key, FormatValue(kvp.Value));
-				}
+				logEvent.MessageTemplate ??= kvp.Value.ToString();
+				return;
+			}
+			var value = FormatValue(kvp.Value);
+			if (!AssignKnownHttpKeys(logEvent, kvp.Key, value))
```

Comment:
This is kind of nice, i.e. if the scope values match the names set by ASP.NET, then st the logEvent.Http schema.

Reply:
@Mpdreamz  maybe update the readme documentation to mention that these are automatically parsed and set.

https://github.com/elastic/ecs-dotnet/blob/main/src/Elasticsearch.Extensions.Logging/README.md